api/controller.go

```diff
@@ -2804,7 +2815,28 @@ func (c *Controller) ActionsListRunsHandler() actionsop.ListRunsHandler {
 				WithPayload(responseErrorFrom(err))
 		}
 
-		after := swag.StringValue(params.After)
+		_, err = deps.Cataloger.GetRepository(deps.ctx, params.Repository)
+		if errors.Is(err, catalog.ErrRepositoryNotFound) {
+			return actionsop.NewListRunsNotFound().
+				WithPayload(responseErrorFrom(catalog.ErrRepositoryNotFound))
+		}
+		if err != nil {
+			return actionsop.NewListRunsDefault(http.StatusInternalServerError).
+				WithPayload(responseErrorFrom(err))
+		}
+		if params.Branch != nil {
+			exists, err := deps.Cataloger.BranchExists(deps.ctx, params.Repository, swag.StringValue(params.Branch))
```

Comment:
Isn't this check cover the Repo check too?

Reply:
true, but you will get branch not found in this case